harmonica/forward/prism.py

```diff
@@ -130,9 +135,29 @@ def prism_gravity(
                 + "mismatch the number of prisms ({})".format(prisms.shape[0])
             )
         _check_prisms(prisms)
+    # Attempt to import numba_progress
+    try:
+        from numba_progress import ProgressBar
+    except ImportError:
+        ProgressBar = None
```

Comment:
Might be better to move this lines outside the function right after we run all the imports in `prism.py`. This way we don't try to import `ProgressBar` on each function call and only do it once when importing harmonica.

Reply:
Opps I thought I already fixed that, thanks!

Reply:
Actually when I move these lines outside the function to the top, the tests failed, and don't raise the `ImportError`. Any suggestions here?

Reply:
```
FAILED ../harmonica/forward/tesseroid.py::harmonica.forward.tesseroid.tesseroid_gravity
FAILED ../harmonica/tests/test_prism.py::test_numba_progress_missing_error - Failed: DID NOT RAISE <class 'ImportError'>
FAILED ../harmonica/tests/test_prism_layer.py::test_numba_progress_missing_error[numpy] - Failed: DID NOT RAISE <class 'ImportError'>
FAILED ../harmonica/tests/test_prism_layer.py::test_numba_progress_missing_error[xarray] - Failed: DID NOT RAISE <class 'ImportError'>
```

Reply:
Nice catch. The problem there is in the test function `@patch` decorator, not with the `prism_gravity` function.
What we want to mock is the `ProgressBar` variable that gets defined in `harmonica.forward.prism` and not the `ProgressBar` class of the `numba_progress` module.

So, in the test function replace the first argument of the `@patch` decorator with `"harmonica.forward.prism.ProgressBar"`. It should look like this:

```python
@patch("harmonica.forward.prism.ProgressBar", None)
def test_numba_progress_missing_error():
    """
    Check if error is raised when progresbar=True and numba_progress package
    is not installed.
    """
    prisms = [
        [-100, 0, -100, 0, -10, 0],
        [0, 100, -100, 0, -10, 0],
        [-100, 0, 0, 100, -10, 0],
        [0, 100, 0, 100, -10, 0],
    ]
    densities = [2000, 3000, 4000, 5000]
    coordinates = [0, 0, 0]
    # Check if error is raised
    with pytest.raises(ImportError):
        prism_gravity(
            coordinates, prisms, densities, field="potential", progressbar=True
        )
```

Reply:
Great! Yeah that makes more sense. I'll update and commit now. Thanks